cumulusci/core/flowrunner.py

```diff
@@ -0,0 +1,428 @@
+        :param step_config: the current step's config (dict from YAML)
+        :param visited_steps: used when called recursively for nested steps, becomes the return value
+        :param parent_options: used when called recursively for nested steps, options from parent flow
+        :param from_flow: used when called recursively for nested steps, name of parent flow
+        :return: List[StepSpec] a list of all resolved steps including/under the one passed in
+        """
+        number = LooseVersion(str(number))
+
+        if visited_steps is None:
+            visited_steps = []
+
+        if parent_options is None:
+            parent_options = {}
+
+        # Step Validation
+        # - A step is either a task OR a flow.
+        if all(k in step_config for k in ("flow", "task")):
+            raise FlowConfigError(
+                "Step {} is configured as both a flow AND a task. \n\t{}.".format(
+                    number, step_config
+                )
+            )
+
+        # Skips
+        # - either in YAML (with the None string)
+        # - or by providing a skip list to the FlowRunner at initialization.
+        if (
+            ("flow" in step_config and step_config["flow"] == "None")
+            or ("task" in step_config and step_config["task"] == "None")
+            or ("task" in step_config and step_config["task"] in self.skip)
+        ):
+            visited_steps.append(
+                StepSpec(
+                    number,
+                    step_config.get("task", step_config.get("flow")),
+                    step_config.get("options", {}),
+                    from_flow=from_flow,
+                    skip=True,  # someday we could use different vals for why skipped
+                )
+            )
+            return visited_steps
+
+        if "task" in step_config:
+            name = step_config["task"]
+
+            # try getting the task, so that we raise an exception for missing tasks here in
+            # preparation rather than at runtime. even though we'll make this same call over
+            # in the TaskRunner.
+            self.project_config.get_task(name)
+
+            step_options = copy.deepcopy(parent_options.get(name, {}))
+            step_options.update(step_config.get("options", {}))
+
+            if name in self.runtime_options:
+                pass
+
+            visited_steps.append(
+                StepSpec(
+                    number,
+                    name,
+                    step_options,
+                    step_config.get("ignore_failure", False),
+                    from_flow=from_flow,
+                )
+            )
+            return visited_steps
+
+        if "flow" in step_config:
+            name = step_config["flow"]
+            step_options = step_config.get("options", {})
+            flow_config = self.project_config.get_flow(name)
+            for sub_number, sub_stepconf in flow_config.steps.items():
```

Comment:
Actually we need to sort by the looseversion here because there's nothing in dictmerge that does that for us. You've lost the sort() from the old _get_steps_ordered

Reply:
great point. I think we can wait til we have the full set of steps to sort it? that was my idea, even if I forgot to code it